### ISSUE TITLE
Testing with magento 2

### ISSUE THREAD

**Issue by Minirock** [magento/magento2#38752]
### Summary

As First let me apologize and say this is not really an issue but more like a search of guidance as all the documentation I've found on this subject was quite dark and somehow obsolete and / or not understandable to my level.

The request is simple by itself : 

When it comes to TDD or more commonly Testing feature with magento 2. What solutions is offered by magento 2 for unit and integration tests ? 

Is using a raw Php Unit the only way to do things ? Don't we have a magento 2 module making test easier ?  It's really becoming complicated when it comes to test changes to database, entity persistence etc etc

### Examples

As an exemple, does some of you have complete solution sample to test things like 

- Expected prices after adding items to cart
- Adding a product 
- Adding a customer
- Adding a custom entity

Basically anything that could lead to alter the database.

Does Magento offers a pre built solution for this kind of test ? If so where can we find the documentation ? 

### Triage and priority

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [X] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by engcom-Bravo** [comment]
Hi @Minirock,

Thanks for your reporting and Collaboration.

We are using this board to address the bugs you are facing in the Magento core library. But it doesn’t seem the same for us. you can raise your queries on [magento.stackexchange](https://magento.stackexchange.com/).

Thank you.
